theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -503,11 +511,13 @@ def _optimize_impl(
             grad_loop_info = self._init_info(
                 track_best_solution, track_err_history, track_state_history
             )
+            detach_hessian = backward_mode == BackwardMode.IMPLICIT
```

Comment:
What do we do in a scenario where I use IMPLCIT and run a problem for only 10 iters but it will take 20 iters to converge to fixed point. Do we default to detach_hessian = true as always with IMPLICIT or reveret to detach_hessian = false?

Reply:
Based on offline discussion plan is to keep the detach true default for IMPLICIT.